Fix klocwork issue

yuanzhel opened the pull request:
Delete an array allocated by new before the function return.

Change-Id: I971c5576a438bac42f9400897ac0fc7ca9df92ce
Tracked-On: https://jira.devtools.intel.com/browse/OAM-85577
Signed-off-by: Yuanzhe Liu <yuanzhe.liu@intel.com>

YuanjunHuang commented:
Use for cherry-pick, need not merge it. Same code will be integrated from next merge bridge.

yuanzhel committed:
Fix klocwork critical issue

Add a null pointer checking before dereference.
Delete an array allocated by new before the function return.

Change-Id: I551f3ed6c417e96dbff159847140eabfdb77b2d7
Test: Boot successful
Tracked-On: No
Signed-off-by: Yuanzhe Liu <yuanzhe.liu@intel.com>

yuanzhel commented:
Add a new kw critical issue fix.
